src/plotSearchAndCompetitionsPage/plotSearchAndCompetitionsPage.tsx

```diff
@@ -72,7 +73,27 @@ const PlotSearchAndCompetitionsPage = (props: Props): JSX.Element => {
     useState<CategoryVisibilities>({});
   const [categoryOptions, setCategoryOptions] = useState<CategoryOptions>([]);
   const [selectedTarget, setSelectedTarget] = useState<SelectedTarget>(null);
+  const [hoveredTargetId, setHoveredTargetId] = useState<number | null>(null);
   const [isSidebarOpen, setSidebarOpen] = useState<boolean>(true);
+  const { id } = useParams();
+
+  useEffect(() => {
+    if (id) {
+      const target = plotSearches
+        .filter((plotSearch) =>
+          plotSearch.plot_search_targets.some((t) => t.id.toString() === id)
+        )
+        .map((p) => ({
+          plotSearch: p,
+          target: p.plot_search_targets.filter(
+            (t) => t.id.toString() === id
+          )[0],
+        }))[0];
+      setSelectedTarget(target);
```

Comment:
Does this redirect to the URL without the ID if the current plot searches didn't include that ID at all?

Reply:
No it doesnt. The unexisting id keeps ghosting at the end of the url until "back-button" is pressed or new target is selected. That doesn't break anything though. But yeah, could be trimmed.